app/utils/tx.ts

```diff
@@ -558,3 +560,30 @@ export function intoParsedTransaction(tx: Transaction): ParsedTransaction {
         signatures: tx.signatures.map(value => bs58.encode(value.signature as any)),
     };
 }
+
+export function intoTransactionErrorReason(info: TransactionStatusInfo, tx?: ParsedTransaction | Transaction): { errorReason: string, errorLink?: string } {
```

Comment:
Can you rename this to `getTransactionErrorReason` & move this back to the page it came from?

It's not sufficiently general to be moved into `utils`